libs/sdk-core/src/breez_services.rs

```diff
@@ -574,6 +574,13 @@ impl BreezServices {
         Ok(self.persister.get_payment_by_hash(&hash)?)
     }
 
+    /// Set the external metadata of a payment as a valid JSON string
+    pub async fn set_payment_metadata(&self, hash: String, metadata: String) -> SdkResult<()> {
```

Comment:
If we require the string to be json then we should validate it is a valid json otherwise we will end up with invalid json in the db.
Also I have noticed int the original issue it was mentioned that it would be nice to be able to search payments for a specific metadata value. This can be done using JSON1 sqlite extension (https://sqlite.org/json1.html). I think we can defer this specific search requirement to a following PR but we need to verify our current solution will be able to supports it.

Reply:
You're right about verifying the JSON. I read that SQLite's `json_set` did checks before inserting, but I just tested out that it is not the case. Regarding searching for specific fields, I believe we can support it right away by exposing a method using `json_extract`.